apps/mobile/src/store/settings/settings.ts

```diff
@@ -0,0 +1,128 @@
+import { useColorScheme } from 'react-native';
+import { useSelector } from 'react-redux';
+
+import { whenTheme } from '@/utils/when-theme';
+
+import {
+  AccountDisplayPreference,
+  BitcoinUnit,
+  DefaultNetworkConfigurations,
+  FiatCurrency,
+  WalletDefaultNetworkConfigurationIds,
+} from '@leather.io/models';
+
+import { useAppDispatch } from '../utils';
+import {
+  selectAccountDisplayPreference,
+  selectAnalyticsEnabled,
+  selectAppAuthEnabled,
+  selectBitcoinUnit,
+  selectConversionUnit,
+  selectNetwork,
+  selectTheme,
+  selectWalletSecurityLevel,
+} from './settings.read';
+import {
+  userChangedAccountDisplayPreference,
+  userChangedBitcoinUnit,
+  userChangedConversionUnit,
+  userChangedNetwork,
+  userChangedTheme,
+  userChangedWalletSecurityLevel,
+  userToggledAnalyticsEnabled,
+  userToggledAppAuthEnabled,
+} from './settings.write';
+
+export const defaultNetworks = ['mainnet', 'testnet', 'signet'] as const;
+
+export const defaultThemes = ['light', 'dark', 'system'] as const;
+export type ThemeStore = (typeof defaultThemes)[number];
+export type Theme = Exclude<ThemeStore, 'system'>;
+
+export type WalletSecurityLevel = 'undefined' | 'secure' | 'insecure';
+
+export interface SettingsState {
+  accountDisplayPreference: AccountDisplayPreference;
+  analyticsEnabled: boolean;
+  appAuthEnabled: boolean;
+  bitcoinUnit: BitcoinUnit;
+  conversionUnit: FiatCurrency;
+  createdOn: string;
+  network: DefaultNetworkConfigurations;
+  theme: ThemeStore;
+  walletSecurityLevel: WalletSecurityLevel;
+}
+
+export const initialState: SettingsState = {
+  accountDisplayPreference: 'ns',
+  analyticsEnabled: false,
+  appAuthEnabled: false,
+  bitcoinUnit: 'bitcoin',
+  conversionUnit: 'USD',
+  createdOn: new Date().toISOString(),
+  network: WalletDefaultNetworkConfigurationIds.mainnet,
+  theme: 'system',
+  walletSecurityLevel: 'undefined',
+};
+
+export function useSettings() {
```

Comment:
Thanks for the refactor. Think this is a really nice pattern where we expose all store read/writes via a uniform hook, abstracting any store-specific implementation details.

Will follow up with a depcruiser rule to prevent components reading `.{read,write}.ts` files